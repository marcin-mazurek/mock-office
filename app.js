const express = require('express');
require('./dist/double-rest-api');

const serve = express.static('./dist/double-gui');
const app = express();
const server = app.use(serve);

server.listen(3070, () => {
  console.log('GUI address: http://127.0.0.1:3070');
});
