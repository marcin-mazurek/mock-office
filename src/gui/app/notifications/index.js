export * from './selectors';
export * from './createNotification';
export { default as reducer } from './reducer';
export * from './actions';
