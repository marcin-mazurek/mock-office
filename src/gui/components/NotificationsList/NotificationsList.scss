.notifications-list {
  @extend %list-reset;
  align-items: center;
  bottom: 20px;
  display: flex;
  flex-direction: column;
  position: absolute;
  width: 100%;
  z-index: 999;
}

.notifications-list__item {
  margin-top: 5px;
}

.notifications-bubble {
  @extend %button-reset;
  border-radius: 22px;
  box-shadow: 0 5px 15px 0 rgba(0, 0, 0, 0.2);
  cursor: pointer;
  color: #FFF;
  font-size: 14px;
  padding: 7px 15px;
  text-align: center;
}

.notifications-bubble--info {
  background-color: $color-info;
}

.notifications-bubble--error {
  background-color: $color-error;
}

.notifications-bubble--success {
  background-color: #23D1A9;
}
