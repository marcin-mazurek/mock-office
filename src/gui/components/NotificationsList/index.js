export { NotificationsList, NOTIFICATION_CLICKED, notificationsClickedAction } from './NotificationsList';
export { default as NotificationsListConnect } from './NotificationsListConnect';
