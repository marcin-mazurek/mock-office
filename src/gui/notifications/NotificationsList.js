import React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Notification } from './createNotification';
import { allNotificationsSelector } from './selectors';
import { removeAction as createRemoveAction } from './actions';

// eslint-disable-next-line import/prefer-default-export
export const NotificationsList = ({ notifications, remove }) => (
  <ul className="notifications-list">
    {
      notifications.map(
        ({ id, text, type }) =>
          <li className="notifications-list-item" key={id}>
            <button
              className={`notifications-bubble notifications-bubble--${type}`}
              onClick={() => remove(id)}
            >
              {text}
            </button>
          </li>
      )
    }
  </ul>
);
NotificationsList.propTypes = {
  notifications: ImmutablePropTypes.listOf(PropTypes.instanceOf(Notification)),
  remove: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  notifications: allNotificationsSelector(state)
});
const mapDispatchToProps = {
  remove: createRemoveAction
};

const NotificationsListConnect = connect(mapStateToProps, mapDispatchToProps)(NotificationsList);

export default NotificationsListConnect;
