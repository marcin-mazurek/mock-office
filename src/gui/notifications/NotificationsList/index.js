export { NotificationsList, NOTIFICATION_CLICKED, notificationsClickedAction } from './NotificationsList';
