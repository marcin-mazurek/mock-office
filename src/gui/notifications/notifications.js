import { SUCCEED as ADD_SERVER_DID_SUCCEED, FAILED as ADD_SERVER_FAILED } from '../servers/addServer/epics';
import { DID_FAIL as EDIT_SERVER_DID_FAIL } from '../servers/editServer/epics';
import {
  DID_FAIL as REMOVE_SERVER_DID_FAIL,
  DID_SUCCEED as REMOVE_SERVER_DID_SUCCEED
} from '../servers/removeServer/epics';
import { DID_FAIL as REMOVE_MOCK_DID_FAIL } from '../mocks/removeMock/epics';
import {
  SUCCEEDED as IMPORT_MOCKS_SUCCEEDED,
  FAILED as IMPORT_MOCKS_FAILED
} from '../mocks/importMock/epics';
import {
  SUCCEED as ADD_HTTP_MOCK_SUCCEED,
  FAILED as ADD_HTTP_MOCK_FAILED
} from '../mocks/addMock/addHttpMock/epic';
import {
  SUCCEED as ADD_WS_MOCK_SUCCEED,
  FAILED as ADD_WS_MOCK_FAILED
} from '../mocks/addMock/addWsMock/epic';
import {
  FAILED as START_SERVER_FAILED
} from '../servers/startServer/epics';
import notificationsReduxModule from './notificationsReduxModule';

const showError = (state, action, module) =>
  module.addNotification(state, { type: 'error', text: action.reason });
const showAddMockSuccess = (state, action, module) =>
  module.addNotification(state, { type: 'success', text: 'Mock added' });

export const reducer = notificationsReduxModule.createReducer({
  [ADD_SERVER_DID_SUCCEED]: (state, action, module) =>
    module.addNotification(state, { type: 'success', text: 'Server added' }),
  [REMOVE_SERVER_DID_SUCCEED]: (state, action, module) =>
    module.addNotification(state, { type: 'success', text: 'Server removed' }),
  [EDIT_SERVER_DID_FAIL]: showError,
  [REMOVE_SERVER_DID_FAIL]: showError,
  [REMOVE_MOCK_DID_FAIL]: showError,
  [IMPORT_MOCKS_FAILED]: showError,
  [ADD_HTTP_MOCK_FAILED]: showError,
  [ADD_WS_MOCK_FAILED]: showError,
  [ADD_SERVER_FAILED]: showError,
  [START_SERVER_FAILED]: showError,
  [IMPORT_MOCKS_SUCCEEDED]: (state, action, module) =>
    module.addNotification(state, { type: 'success', text: 'Mocks imported' }),
  [ADD_HTTP_MOCK_SUCCEED]: showAddMockSuccess,
  [ADD_WS_MOCK_SUCCEED]: showAddMockSuccess
});
export const selectors = notificationsReduxModule.createSelectors(state => state.get('notifications'));
