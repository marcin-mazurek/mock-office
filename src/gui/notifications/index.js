import {
  SUCCEEDED as ADD_SERVER_DID_SUCCEED,
  FAILED as ADD_SERVER_FAILED
} from '../servers/addServer/epics';
import { FAILED as EDIT_SERVER_DID_FAIL } from '../servers/editServer/epics';
import {
  FAILED as REMOVE_SERVER_DID_FAIL,
  SUCCEEDED as REMOVE_SERVER_DID_SUCCEED
} from '../servers/removeServer/epics';
import { DID_FAIL as REMOVE_MOCK_DID_FAIL } from '../mocks/removeMock/epics';
import {
  SUCCEEDED as IMPORT_MOCKS_SUCCEEDED,
  FAILED as IMPORT_MOCKS_FAILED
} from '../mocks/importMock/epics';
import {
  SUCCEED as ADD_HTTP_MOCK_SUCCEED,
  FAILED as ADD_HTTP_MOCK_FAILED
} from '../modules/addHttpMock/epic';
import {
  SUCCEED as ADD_WS_MOCK_SUCCEED,
  FAILED as ADD_WS_MOCK_FAILED
} from '../modules/addWsMock/epic';
import {
  FAILED as START_SERVER_FAILED
} from '../servers/startServer/startServer';
import { NOTIFICATION_CLICKED } from './NotificationsList';
import notificationsReduxModule from './notificationsReduxModule';
import createNotificationsListConnect from './createNotificationsListConnect';
import { FAILED as STOP_SERVER_FAILED } from '../servers/stopServer';
import { NOTIFICATION_EXPIRED } from './clearExpiredNotificationsEpic';

const showError = (state, action, reducers) =>
  reducers.addNotification(state, { type: 'error', text: action.reason });
const showAddMockSuccess = (state, action, reducers) =>
  reducers.addNotification(state, { type: 'success', text: 'Mock added' });

const notifications = notificationsReduxModule(
  {
    [ADD_SERVER_DID_SUCCEED]: (state, action, reducers) =>
      reducers.addNotification(state, { type: 'success', text: 'Server added' }),
    [REMOVE_SERVER_DID_SUCCEED]: (state, action, reducers) =>
      reducers.addNotification(state, { type: 'success', text: 'Server removed' }),
    [EDIT_SERVER_DID_FAIL]: showError,
    [REMOVE_SERVER_DID_FAIL]: showError,
    [REMOVE_MOCK_DID_FAIL]: showError,
    [IMPORT_MOCKS_FAILED]: showError,
    [ADD_HTTP_MOCK_FAILED]: showError,
    [ADD_WS_MOCK_FAILED]: showError,
    [ADD_SERVER_FAILED]: showError,
    [START_SERVER_FAILED]: showError,
    [STOP_SERVER_FAILED]: showError,
    [IMPORT_MOCKS_SUCCEEDED]: (state, action, reducers) =>
      reducers.addNotification(state, { type: 'success', text: 'Mocks imported' }),
    [ADD_HTTP_MOCK_SUCCEED]: showAddMockSuccess,
    [ADD_WS_MOCK_SUCCEED]: showAddMockSuccess,
    [NOTIFICATION_CLICKED]: (state, action, reducers) =>
      reducers.removeNotification(state, action.id),
    [NOTIFICATION_EXPIRED]: (state, action, reducers) =>
      reducers.removeNotification(state, action.id)
  },
  state => state.get('notifications')
);

export const reducer = notifications.reducer;
export const selectors = notifications.selectors;
export const NotificationsListConnect = createNotificationsListConnect(notifications.selectors);
