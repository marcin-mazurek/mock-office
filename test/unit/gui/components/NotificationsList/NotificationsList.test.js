import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';
import enzymeToJson from 'enzyme-to-json';
import { NotificationsList } from '../../../../../src/gui/components/NotificationsList';
import { createNotification } from '../../../../../src/gui/app/notifications';

describe('NotificationsList', () => {
  describe('snapshots', () => {
    test('without notifications', () => {
      const props = {
        notifications: new List(),
        onNotificationClick: () => {},
        onNewNotificationDisplayed: () => {}
      };
      const wrapper = shallow(<NotificationsList {...props} />);
      expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    test('with notification', () => {
      let notification = createNotification({ text: 'message' });
      notification = notification.set('id', 'some-id');
      const props = {
        notifications: new List([
          notification
        ]),
        onNotificationClick: () => {},
        onNewNotificationDisplayed: () => {}
      };
      const wrapper = shallow(<NotificationsList {...props} />);
      expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
  });
});
