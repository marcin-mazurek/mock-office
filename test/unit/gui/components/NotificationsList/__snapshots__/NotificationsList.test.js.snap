// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsList snapshots with notification 1`] = `
<ul
  className="notifications-list"
>
  <li
    className="notification-list__item"
  >
    <NotificationListItem
      id="cj77cllps0000rjvehhwuhwyc"
      onNotificationClick={[Function]}
      text="message"
      type="info"
    />
  </li>
</ul>
`;

exports[`NotificationsList snapshots without notifications 1`] = `
<ul
  className="notifications-list"
/>
`;
